src/bls12_381/fr.rs

```diff
@@ -561,8 +562,7 @@ impl SqrtField for Fr {
             return Some(*self);
         }
 
-        // if self^((r - 1) // 2) != 1
-        if self.pow([0x7fffffff80000000, 0xa9ded2017fff2dff, 0x199cec0404d0ec02, 0x39f6d3a994cebea4]) != Self::one() {
+        if let QNonResidue = self.legendre() {
```

Comment:
Why is this ``if let QNonResidue =``? (Maybe I'm just missing something about Rust pattern matching, but I would have expected ``if self.legendre() != QResidue`` for similarity to the existing code. I know the zero case is handled above.)